lib/endpoints/AuthorizationCodeFlowEndpoint.js

```diff
@@ -1,175 +1,121 @@
- * ('/' by default).
- * @param {string?} endpointConfig.cookie.domain Domain attribute for cookie.
- * @constructor
- * @extends GrantFlowMiddlewareBase
- */
-function AuthorizationCodeFlowEndpoint(
-		$serviceLocator, authConfig, endpointConfig
-	) {
-	GrantFlowMiddlewareBase.call(this, $serviceLocator, endpointConfig);
-	this._sender = new AuthorizationCodeSender(
-		$serviceLocator, authConfig
-	);
-	if (typeof (endpointConfig.redirectUri) !== 'string' ||
-		endpointConfig.redirectUri.length === 0) {
-		throw new Error(ERROR_REDIRECT_URI);
-	}
-	this._redirectUri = endpointConfig.redirectUri;
+class AuthorizationCodeFlowEndpoint extends GrantFlowMiddlewareBase {
 
-	if (typeof (endpointConfig.returnUri) !== 'string' ||
-		endpointConfig.returnUri.length === 0) {
-		throw new Error(ERROR_RETURN_URI);
-	}
-	this._returnUri = new URI(endpointConfig.returnUri);
-}
+	/**
+	 * Creates new instance of authorization code flow endpoint.
+	 * @param {ServiceLocator} $serviceLocator Service locator to resolve
+	 * dependencies.
+	 *
+	 * @param {Object} authConfig Authorization configuration.
+	 * @param {string} authConfig.clientId Id of OAuth 2.0 client.
+	 * @param {string} authConfig.clientSecret Secret of OAuth 2.0 client.
+	 * @param {string} authConfig.authServerUrl URL to OAuth 2.0
+	 * authentication server.
+	 * @param {number?} authConfig.timeout Grant send timeout (30000 ms by default).
+	 * @param {boolean?} authConfig.unsafeHTTPS Determines if sender will send grant
+	 * via unsafe HTTPS connection with invalid certificate (false by default).
+	 * @param {string?} authConfig.tokenEndpointPath URL path to
+	 * OAuth 2.0 token endpoint (/token by default).
+	 *
+	 * @param {Object} endpointConfig Endpoint configuration.
+	 * @param {string?} endpointConfig.scope Access scope for endpoint.
+	 * @param {string} endpointConfig.redirectUri Redirect URI used for
+	 * obtaining code.
+	 * @param {string} endpointConfig.returnUri Return URI used to redirect
+	 * user agent after authorization.
+	 * @param {Object} endpointConfig.cookie Token cookie configuration.
+	 * @param {string} endpointConfig.cookie.accessTokenName Name of cookie
+	 * with access token.
+	 * @param {string?} endpointConfig.cookie.accessTokenExpiresIn Expiration time
+	 * in seconds for access token cookie if it is not specified by authorization
+	 * server (3 600 secs by default, 1 hour).
+	 * @param {string} endpointConfig.cookie.refreshTokenName Name of cookie
+	 * with refresh token.
+	 * @param {string?} endpointConfig.cookie.refreshTokenExpiresIn Expiration time
+	 * in seconds for refresh token cookie
+	 * (3 110 400 000 secs by default, 100 years).
+	 * @param {string?} endpointConfig.cookie.path Path attribute for cookie
+	 * ('/' by default).
+	 * @param {string?} endpointConfig.cookie.domain Domain attribute for cookie.
+	 *
+	 * @extends GrantFlowMiddlewareBase
+	 */
+	constructor($serviceLocator, authConfig, endpointConfig) {
```

Comment:
Let's remove a `$` character, because it can be confusing, people can think that it's still a DI.